### PR TITLE
ui testing HW

### PR DESCRIPTION
**Pull request by VeraZh2022**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,19 @@
 {
-  "name": "practiceproject-in-epam-automated-course",
+  "name": "wdio-v8",
   "version": "1.0.0",
-  "description": "Automated testing in JavaScript",
+  "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (andrxo) [inline package.json]: Pls remove unused script(s)

File: package.json [hunk 2]
```diff
@@ -1,12 +1,19 @@
 {
-  "name": "practiceproject-in-epam-automated-course",
+  "name": "wdio-v8",
   "version": "1.0.0",
-  "description": "Automated testing in JavaScript",
+  "description": "",
```
Comment (andrxo) [inline package.json]: Why have you removed this description ?

File: src/config/wdio.conf.js [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    //
+    // If you only want to run your tests until a specific amount of tests have failed use
+    // bail (default is 0 - don't bail, run all tests).
+    bail: 0,
+    //
+    // Set a base URL in order to shorten url command calls. If your `url` parameter starts
+    // with `/`, the base url gets prepended, not including the path portion of your baseUrl.
+    // If your `url` parameter starts without a scheme or `/` (like `some/path`), the base url
+    // gets prepended directly.
+    baseUrl: 'http://localhost',
```
Comment (andrxo) [inline src/config/wdio.conf.js]: Pls retup a real URL

File: package.json (hunk 1 above)
Comment (VeraZh2022) [reply package.json]: Done.

File: package.json (hunk 2 above)
Comment (VeraZh2022) [reply package.json]: Added back.

File: src/config/wdio.conf.js (hunk 3 above)
Comment (VeraZh2022) [reply src/config/wdio.conf.js]: Real baseUrl is set up.

File: ui-tests/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "wdio-v8",
+  "version": "1.0.0",
+  "description": "Automated testing in JavaScript",
+  "main": "index.js",
```
Comment (andrxo) [inline ui-tests/package.json:5]: Pls remove it

File: ui-tests/src/tests/advanced.tests.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+describe('Patient list page', () => {
+
+    it('Adding new patient using waitUntil()', async () => {
+        await browser.url('#/patients');
+
+        const lastRow = await $('//table[@class="e-table"]//tr[last()]/td[@class="e-rowcell"]');
+        const lastRowIndex = await lastRow.getAttribute('index');
+
+        const btnAddNewPatient = await $('[class="e-control e-btn e-lib e-normal add-details e-primary"]');
+        const btnSaveNewPatient = await $('//ejs-dialog[contains(@class, "new-patient-dialog")]//div[@class="e-footer-content"]//button[text()="Save"]');
```
Comment (andrxo) [inline ui-tests/src/tests/advanced.tests.js:39]: pls fix this selector and all others

File: ui-tests/src/tests/advanced.tests.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        await browser.waitUntil(async () => {
+            const newRow = await $('//table[@class="e-table"]//tr[last()]/td[@class="e-rowcell"]');
+            newRowIndex = await newRow.getAttribute('index');
+            return newRowIndex > lastRowIndex;
+        }, {
+            timeout: 10000,
+            timeoutMsg: 'New row is not added'
+        });
+        // Validation
+        expect(newRowIndex).toHaveValue(lastRowIndex + 1);
```
Comment (andrxo) [inline ui-tests/src/tests/advanced.tests.js:71]: .toBe

Comment (andrxo) [reply ui-tests/src/tests/advanced.tests.js:71]: a redundant verification

File: ui-tests/src/tests/advanced.tests.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+});
+
+describe('Doctor list page', async () => {
+
+    it('Changing doctor break time in combobox by typing', async () => {
+        //open doctor page
+        await browser.url('#/doctors');
+        //click 'Select a Specialization' dropdown list
+        const selectSpecialization = await $('//div[@class="specialization-types"]//ejs-dropdownlist[@id="Specialization"]');
+        browser.action('pointer')
```
Comment (andrxo) [inline ui-tests/src/tests/advanced.tests.js:82]: missed await

Comment (andrxo) [reply ui-tests/src/tests/advanced.tests.js:82]: the same for all other 'actions'

File: ui-tests/src/tests/advanced.tests.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+        ]);
+        await browser.pause(1000);
+        //click Save button 
+        const btnSaveBreakHours = await $('//ejs-dialog[contains(@class, "break-hour-dialog")]//div[@class="e-footer-content"]//button[text()="Save"]');
+        await btnSaveBreakHours.click();
+        //'Break hours' window is closed
+        await expect($('ejs-dialog[class$="e-popup-open"]')).not.toBeDisplayed();  
+        await browser.pause(1000); 
+        //doctor details page is opened with changed data 
+        await expect($('//div[contains(@class, "day-break-hours")]')).toHaveText('3:30 PM - 4:30 PM');
```
Comment (andrxo) [inline ui-tests/src/tests/advanced.tests.js:150]: div.day-break-hours

File: ui-tests/src/tests/local.storage.tests.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    it('Check data removing from local storage', async () => {
+        await browser.execute(() => {
+            localStorage.setItem('Time', '10:30 AM, 01:00 PM, 04:00 PM, 07:00 PM');
+        });
+
+        //data existing validation
+        const newData = await browser.execute(() => {
+            return localStorage.getItem('Time');
+        });
```
Comment (andrxo) [inline ui-tests/src/tests/local.storage.tests.js:37]: const newData = await browser.execute(() => localStorage.getItem('Time'));

File: ui-tests/src/tests/advanced.tests.js (hunk 8 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/advanced.tests.js:150]: +

File: ui-tests/package.json (hunk 4 above)
Comment (VeraZh2022) [reply ui-tests/package.json:5]: done

File: ui-tests/src/tests/advanced.tests.js (hunk 8 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/advanced.tests.js:150]: done

File: ui-tests/src/tests/advanced.tests.js (hunk 6 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/advanced.tests.js:71]: done

File: ui-tests/src/tests/advanced.tests.js (hunk 5 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/advanced.tests.js:39]: done

File: ui-tests/src/tests/advanced.tests.js (hunk 7 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/advanced.tests.js:82]: done

File: ui-tests/src/tests/local.storage.tests.js (hunk 9 above)
Comment (VeraZh2022) [reply ui-tests/src/tests/local.storage.tests.js:37]: done

**Pull request merged**